src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int maxParameterValue = 50;
+    private static final int maxFigureNumber = 5;
+    private static final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        double randomParameter = new Random().nextInt(maxParameterValue);
+        int value = random.nextInt(maxFigureNumber);
+        switch (value) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(), randomParameter);

```

Comment:
Don't pass expressions in the constructor, instead create variables. See [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor).